cpu/sam0_common/periph/flashpage.c

```diff
@@ -103,7 +103,16 @@ void flashpage_write(int page, const void *data)
 
     /* write data to page */
     if (data != NULL) {
-        flashpage_write_raw(page_addr, data, FLASHPAGE_SIZE);
+        /* One RIOT page is FLASHPAGE_PAGES_PER_ROW SAM0 flash pages (a row) as
+         * defined in the file cpu/sam0_common/include/cpu_conf.h, therefore we
+         * have to split the write into FLASHPAGE_PAGES_PER_ROW raw calls
+         * underneath, each writing a physical page in chunks of 4 bytes (see
+         * flashpage_write_raw)
+         * The erasing is done once as a full row is always reased.
+         */
+        for (unsigned curpage = 0; curpage < FLASHPAGE_PAGES_PER_ROW; curpage++) {
+            flashpage_write_raw(page_addr + (curpage * NVMCTRL_PAGE_SIZE / 4), (void *) ((uint32_t *) data + (curpage * NVMCTRL_PAGE_SIZE / 4)), NVMCTRL_PAGE_SIZE);
```

Comment:
Please stay within the 80 characters limit per lines. @aabadie will murder me if we merge this ;)

Reply:
Just one question: is 80 very strict? To keep it nicely readadble I would keep it as:

```
        for (unsigned curpage = 0; curpage < FLASHPAGE_PAGES_PER_ROW; curpage++) {
            flashpage_write_raw(page_addr + (curpage * NVMCTRL_PAGE_SIZE / 4),
                                (void *) ((uint32_t *) data + (curpage * NVMCTRL_PAGE_SIZE / 4)),
                                NVMCTRL_PAGE_SIZE);
        }
```
But the line with the second parameter is 98 sadly. Otherwise I can split that too to make it <= 80 but will become somehow less readable, something like:
```
        for (unsigned curpage = 0; curpage < FLASHPAGE_PAGES_PER_ROW; curpage++) {
            flashpage_write_raw(page_addr + (curpage * NVMCTRL_PAGE_SIZE / 4),
                                (void *) ((uint32_t *) data + 
                                (curpage * NVMCTRL_PAGE_SIZE / 4)),
                                NVMCTRL_PAGE_SIZE);
        }
```

Reply:
Having it less than 100 is ok. The most important is to keep the code readable.

Reply:
Thanks for the indication! I committed now the first version which is much more readable and still better than the original long line. Hope it's fine now!